swupd/rename_test.go

```diff
@@ -0,0 +1,151 @@
+package swupd
+
+import (
+	"os"
+	"syscall"
+	"testing"
+)
+
+// Need to set up a type to hold the FileInfo
+// as we are not holding them on disk
+type mockinfo struct {
+	os.FileInfo // Embedded but only used to get things like Name() method
+	size        int64
+}
+
+// Here is the method to override the FileInfo.Size() for mockinfo
+func (m mockinfo) Size() int64 {
+	return m.size
+}
+
+// Constructor for os.FileInfo using mockinfo
+func sizer(s int64) os.FileInfo {
+	return mockinfo{size: s}
+}
+
+// generateTestArray creates some File structures with shortnames and
+// returns a map of them.
+func generateTestArray(t *testing.T) map[string]File {
+	// Set up an array of File with short names
+	const (
+		_rwx = syscall.S_IFREG + 0644
+	)
+	files := []struct {
+		n string // short name for the tests
```

Comment:
This ended up being confusing, because there's also the notion of "shortname" above that is the name without digits.

Optional: I think the indirection here doesn't add much. I would explore one of those two venues.

(a) Instead of having generateTestArray, have a "createTestFile" (with a better name), that takes the parameters and creates a *File (or something else) with the properties you want. Then in the top of the test you create those assigning variable names that could act as your short name. The arrays would use those variables instead of strings.
(b) Use the real filenames in the test. The testcase entries will be multiline, but I think its fine.

Reply:
My intention with the generateTestArray is to provide a pool of names that tests can draw on, rather than having each test create their own. I look at some of the manifest tests for example and it seems like 90% of the test is setup. It probably needs more tests to be written to make the advantage clear.

Perhaps I should set the field which has the name without digits to be partialName, which fits in better with another use case for the field as well.

Initially I did use the full names in the tests, but they made the tests long enough that my eyes started glazing over.  When you have several similar but different names I found the L1, L2 type short names much easier to reason about.